src/index.js

```diff
@@ -0,0 +1,89 @@
+import { InstanceBase, runEntrypoint } from '@companion-module/base'
+import { initActions } from './actions.js'
+import { API } from './api.js'
+import { initFeedbacks } from './feedbacks.js'
+import { initProduct } from './models.js'
+import { setPolling } from './polling.js'
+import { initPresets } from './presets.js'
+import UpgradeScripts from './upgrades.js'
+import { initVariables, checkVariables } from './vars.js'
+import { getConfigDefinitions } from './config.js'
+
+class PTZControlerInstance extends InstanceBase {
+	constructor(internal) {
+		super(internal)
+		this.data = {}
+		this.polling = {}
+		this.product
+	}
+
+	async init(config) {
+		this.data = {
+			camera: 'NaN',
+			group: 'NaN',
+			port: 'NaN',
+		}
+
+		this.config = config
+		this.config.model = this.config.model || 'AW-RP50'
+		this.config.enablePolling = this.config.enablePolling || true
+
+		this.updateStatus('connecting')
+		this.api = new API(config)
+		this.configUpdated(config)
+		this.updateStatus('ok')
+	}
+
+	async destroy() {
+		if (this.socket !== undefined) {
+			this.socket.destroy()
+		}
+
+		if (this.polling.interval) {
+			clearInterval(this.polling.interval)
+		}
+
+		this.debug('destroy', this.id)
+	}
+
+	async configUpdated(config) {
+		const self = this
+
+		self.config = config
+		this.updateStatus('connecting')
+		this.updateStatus('unknown_error')
+
+		// Update API only if config changes are relevant
+		if (config.host != self.config.host || config.httpPort != self.config.httpPort) {
+			self.api.updateConfig(config)
+		}
+		// Set polling only if config changes are relevant
+		if ((config.enablePolling && !this.polling.interval) ||
+			config.apiPollInterval != self.config.apiPollInterval ||
+			config.enablePolling != self.config.enablePolling) {
+            	setPolling(this)
+        }
+
+		self.config = config
+		self.product = initProduct(self.config.model)
+		
+		initActions(this)
+		initFeedbacks(this)
+		self.checkFeedbacks()
+		initPresets(this)
+		initVariables(this)
+		checkVariables(this)
+
+		if (self.config.host) {
+			this.updateStatus('ok')
```

Comment:
The existance of a valid ip does not mean the status is ok.

Reply:
I agree with you in principle. This behavior seems to have been introduced in [this commit](https://github.com/bitfocus/companion-module-panasonic-camera-controller/commit/fe3d403c7f3ddb18c53c6efc546e2bfbda28f56a). As my work has mostly consisted in upgrading the module for Companion v3 without changing the code logic, I kept it. I can work on a new status management for the module, if you want.

Reply:
Yes, but Companion v3 offers more status choices then previous versions. So I think it is a good habit to take this into account when upgrading to v3. Most modules follow the rule to give ok unless there has been an error, i.e. unknown results in ok too. We have now better ways to reflect this.

Reply:
Lasts commits changed the way status is updated